src/systemd-sonic-generator/systemd-sonic-generator.c

```diff
@@ -569,7 +569,11 @@ int get_num_of_asic() {
                     str_num_asic = strtok_r(NULL, "=", &saveptr);
                     strip_trailing_newline(str_num_asic);
                     if (str_num_asic != NULL){
-                        sscanf(str_num_asic, "%d",&num_asic);
+                        char *endptr;
+                        num_asic = strtol(str_num_asic, &endptr, 10);
+                        if (*endptr != '\0') {
+                            fprintf(stderr, "Failed to get num asics.\n");
```

Comment:
>fprintf [](http://example.com/codeflow?start=28&length=7)

You will treat any extra chars after num as failure. I am thinking original sscanf just ignore them, and it's safe to ignore them using strtol.

How about just checking `num_asic > 0` ?